pygeo/mphys/mphys_dvgeo.py

```diff
@@ -205,14 +225,22 @@ def nom_getDVGeo(self, childName=None, DVGeoName=None):
         else:
             DVGeo = self.DVGeos["defaultDVGeo"]
 
-        # return the top level DVGeo
-        if childName is None:
-            return DVGeo
+        # return a child of a DVGeoMulti component
+        if childName is not None and comp is not None:
+            return DVGeo.DVGeoDict[comp].children[childName]
 
-        # return a child DVGeo
-        else:
+        # return a component of a DVGeoMulti
+        elif comp is not None:
+            return DVGeo.DVGeoDict[comp]
+
+        # return a child of a DVGeo
+        elif childName is not None:
             return DVGeo.children[childName]
 
+        # return the top level DVGeo
```

Comment:
I agree with Anil on using this to cover finding the right DVGeo even with the child/component logic instead of repeating the same checks. I think I just didn't realize it could eliminate the checks for child that I put in when I redid the DV adding functions.

Reply:
With the recent changes to DVGeo, someone took care of this update already. All the checks should be streamlined now.